psh/tools/psh.py

```diff
@@ -0,0 +1,99 @@
+'''psh module with tools for psh related tests'''
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh module with tools for psh related tests
+#
+# Copyright 2021 Phoenix Systems
+# Author: Jakub Sarzyński, Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import re
+import pexpect
+
+
+from trunner.config import CURRENT_TARGET, DEVICE_TARGETS
+
+
+EOL = r'(\r+)\n'
+PROMPT = r'(\r+)\x1b\[0J' + r'\(psh\)% '
+
+
+def init(pexpect_proc):
+    ''' Runs psh and asserts a first prompt'''
+    run(pexpect_proc)
+    assert_only_prompt(pexpect_proc)
+
+
+def _readable(exp_regex):
+    ''' Gets more readable expected regex with new lines instead of raw string EOL,
+    and prompt without raw esc sequences. '''
+    exp_regex = exp_regex.replace(EOL, '\n')
+    return exp_regex.replace(PROMPT, '(psh)% ')
+
+
+def assert_cmd(pexpect_proc, cmd, expected, msg='', is_regex=False):
+    ''' Sends specified command and asserts that it's displayed correctly
+    with optional expected output and next prompt'''
+    pexpect_proc.sendline(cmd)
+    cmd = re.escape(cmd)
+    exp_regex = ''
+    if type(expected) is tuple:
+        for line in expected:
+            if not is_regex:
+                line = re.escape(line)
+            exp_regex = exp_regex + line + EOL
+    else:
+        exp_regex = expected
+        if not is_regex:
+            exp_regex = re.escape(exp_regex)
+        exp_regex = exp_regex + EOL
```

Comment:
```suggestion
    if not isinstance(expected, tuple):
        expected = (expected)
    for line in expected:
        if not is_regex:
            line = re.escape(line)
        exp_regex += line + EOL
```

Reply:
It looks better, but when `expected = (expected)` is used particular letters are treated as tuple items (not whole regex as one item, which will work here).

Reply:
Yeap I forgott that comma determines tuples, not parentheses. `expected = tuple((expected,))` should work

Reply:
Oh right `(expected,)` works too, but I prefer `expected = tuple((expected,))` for better readability.